controllers/growth.js

```diff
@@ -3,45 +3,208 @@
+
+        if (result.length) {
+            from = moment(result[0].linked_date).format('YYYY-MM-DD');
+            to = moment(result[0].linked_date).add(2, 'month').format('YYYY-MM-DD');
+        }
+
+        mysql.open(config.DB)
+            .query(
+                'SELECT * FROM channel_stats WHERE channel_id = ? '
+                + 'AND insert_date BETWEEN ? AND ? ORDER BY insert_date',
+                [data.id, from, to],
+                send_response
+            )
+            .end();
+    }
+
+    function send_response (err, result) {
+        if (err) {
+            return next(err);
+        }
+
+        if (!result.length) {
+            return res.warn(404, 'No channel statistics');
+        }
+
+        res.send(result);
+    }
+
+    start();
+};
+
+
+
+exports.get_grouped_channel = function (req, res, next) {
+    var date = moment(),
+        start_month = date.subtract(3, 'month').format('YYYY-MM'),
+        end_month = date,
+        data = util.get_data([], ['month', 'network',
+                             'duration'], req.query),
+        channels = [],
+        cache_key,
+        cache = new memcached(config.memcached.nodes);
+
+    function start () {
+        var simple_key;
+
+        if (data.month) {
+            end_month = moment(data.month + '-01').format('YYYY-MM');
+        }
+
+        if (+data.duration) {
+            start_month = moment(end_month + '-01')
+                .subtract(+data.duration, 'month').format('YYYY-MM');
+        }
+
+        console.log('from ' + start_month + ' to ' + end_month);
+        console.log(data.network);
+
+        simple_key = data.network.split(',') + ':' + start_month + ':' + end_month;
+        cache_key = crypto.createHash('md5').update(simple_key).digest('hex');
 
-            mysql.open(config.DB)
-                .query(
-                    query, [data.id, from, to],
-                    send_response
+        // check cache first
+        cache.get(cache_key, retrieve_cache);
+    }
+
+    function retrieve_cache (err, cache_result) {
+
+        if (!err || !cache_result) {
+
+            // retrieve list of channels
+            return mysql.open(config.DB)
+                .query('SELECT SQL_CACHE channel_id, title, linked_date '
+                       + 'FROM mcn_channels '
+                       + 'WHERE network IN (?) '
+                       + 'AND DATE_FORMAT(linked_date, "%Y-%m") = ?',
+                       [data.network.split(','), start_month],
+                       get_stat
                 )
                 .end();
-        },
+        }
+
+        return res.send(cache_result);
+    }
+
+    function get_stat (err, result) {
+        var channel_ids = [];
+
+        if (err) {
+            return next(err);
+        }
+
+        if (!result.length) {
+            return res.send({ results: [] });
+        }
+
+        channels = result;
+
+        channels.forEach(function (row) {
+            channel_ids.push(row.channel_id);
```

Comment:
if this is just a one time use var, you can just use `Array.map` when using it in the mysql call